File: blockchain/src/main/java/pt/um/lei/masb/blockchain/Sizeable.java
package pt.um.lei.masb.blockchain;

import org.openjdk.jol.info.GraphLayout;

/**
 * Report an approximate size in bytes of the underlying object.
 */
public interface Sizeable {
    default long getApproximateSize() {
       return GraphLayout.parseInstance(this).totalSize();
    }
}

File: blockchain/src/main/kotlin/pt/um/lei/masb/blockchain/Sizeable.kt
package pt.um.lei.masb.blockchain

import org.openjdk.jol.info.GraphLayout

/**
 * Report an approximate size in bytes
 * of the underlying object.
 */
interface Sizeable {
    val approximateSize: Long
        get() = GraphLayout
            .parseInstance(this)
            .totalSize()
}
